classes/module/Module.php

```diff
@@ -1378,9 +1378,12 @@ public static function getModulesOnDisk($use_config = false, $logged_on_addons =
                         $errors[] = Context::getContext()->getTranslator()->trans('%1$s (parse error in %2$s)', [$module, substr($file_path, strlen(_PS_ROOT_DIR_))], 'Admin.Modules.Notification');
                     }
 
-                    $ns = preg_replace('/\n[\s\t]*?namespace\s/', '', $file);
-                    $ns = rtrim($ns, ';');
-                    $module = $ns . '\\' . $module;
+                    preg_match('/\n[\s\t]*?namespace\s.*?;/', $file, $ns);
+                    if (!empty($ns)) {
```

Comment:
@PierreRambaud May be add a comment for explaining why it's not redundant.

Reply:
It's logic the preg_match catches a data when the preg_replace replaces something.
And after if you check the condition, the result of $ns is added even if there is no namespace.